src/runtime/write_debug_image.cpp

```diff
@@ -108,16 +108,19 @@ WEAK bool ends_with(const char *filename, const char *suffix) {
 }
 
 struct ScopedFile {
-    void *f;
-    ALWAYS_INLINE ScopedFile(const char *filename, const char *mode) {
-        f = fopen(filename, mode);
+    void *const user_context;
+    void *const f;
+    ALWAYS_INLINE ScopedFile(void *user_context, const char *filename, const char *mode)
+        : user_context(user_context), f(fopen(filename, mode)) {
+        halide_msan_annotate_memory_is_initialized(user_context, &f, sizeof(f));
```

Comment:
Please add a comment explaining that args to a syscall that live on the stack must be annotated when msan is on. I guess the filename and mode always come from .rodata? Otherwise this code is baffling.

Reply:
I don't think that's the problem. The filename and mode come from code with msan enabled, so they are annotated correctly. The problem here is that fopen wasn't built with msan, so the returned value looks uninitialized to msan.